docs/deployer/healthchecks.md

```diff
@@ -0,0 +1,112 @@
+    # required if no resourceSelector is specified, can be combined with a resourceSelector which is potentially harmful
+    labelSelector:
+      apiVersion: apps/v1
+      kind: Deployment
+      matchLabels:
+        app: myApp
+        component: backendService
+    # requirements specifies what condition must hold true for the given objects to pass the health check
+    # multiple requirements can be given and they all need to successfully evaluate
+    requirements:
+    # jsonPath denotes the path of the field of the selected object to be checked and compared
+    - jsonPath: .status.readyReplicas
+      # operator specifies how the contents of the given field should be compared to the desired value
+      # allowed operators are: DoesNotExist(!), Exists(exists), Equals(=, ==), NotEquals(!=), In(in), NotIn(notIn)
+      operator: In
+      # values is a list of values that the field at jsonPath must match to according to the operators
+      values:
+      - value: 1
+      - value: 2
+      - value: 3
+```
+
+## Default health checks
+
+If the default health check is enabled for a DeployItem (which is the default), the following native Kubernetes resources will be checked as follows:
+
+* `Pod`: It is considered healthy if it successfully completed
+or if it has the the PodReady condition set to true.
+* `Deployment`: It is considered healthy if the controller observed
+its current revision and if the number of updated replicas is equal
+to the number of replicas.
+* `ReplicaSet`: It is considered healthy if its controller observed
+its current revision and if the number of updated replicas is equal to the number of replicas.
+* `StatefulSet`: It is considered healthy if its controller observed
+its current revision, it is not in an update (i.e. UpdateRevision is empty)
+and if its current replicas are equal to its desired replicas.
+* `DaemonSet`: It is considered healthy if its controller observed
+its current revision and if its desired number of scheduled pods is equal
+to its updated number of scheduled pods.
+* `ReplicationController`: It is considered healthy if its controller observed
+its current revision and if the number of updated replicas is equal to the number of replicas.
+
+## Custom Health Checks
+
+Custom health checks can be used to match custom fields of selected resources to given values.
+
+Multiple custom health checks can be defined for DeployItem as multiple resources to be checked each require a specifially tailored custom health check.
+
+A custom health check must contain a name and a selector that selects the resource to be checked. This selector can either be:
+
+- a set of `resourceSelector` each of which matches just one resource at a time, identified by its `apiVersion`, `kind`, `namespace` and `name`
+- a `labelSelector` that matches multiple resources of the same `apiVersion` and `kind`, identified by a set of labels they need to have
```

Comment:
I think we should mention that only resources that are deployed by the deployitem are matched.
Otherwise people might think that with a label selector all resources in the cluster are checked.

Reply:
That piece of information was a bit hidden, it is now explicitly mentioned (twice!).